### DISCUSSION TITLE
Does DepotDownloader delete files missing from filesList?

### DISCUSSION THREAD

**Discussion by ryantheleach** [SteamRE/DepotDownloader#315] [Q&A] [answered]
I've been experimenting with DepotDownloader, and I think (but havn't verified) that DepotDownloader deletes files that are not listed in filesList that are included on the manifest, but are not present in the filesList.

Is this true?

**Comment by azuisleet** [comment, marked as answer]
Files are only deleted if we have a previous manifest to compare to.

If a file appeared previously in an older manifest, and it would have been included in the filelist, then it will be deleted if it does not appear in the new manifest.

If the file exists on disk, but it did not match the filelist, it would not be considered for deletion.

https://github.com/SteamRE/DepotDownloader/blob/master/DepotDownloader/ContentDownloader.cs#L1035

**Comment by ryantheleach** [comment]
Does it store the old manifests in the directory specified by -dir? or is there a global cache somewhere. I'm inspecting old releases, where a file has appeared and been missing several times, presumably by error (by the developer/publisher) and had been outputting the various manifest versions to separate directories.

**Reply by azuisleet** [reply]
DepotDownloader stores the currently installed manifest ID in `.DepotDownloader/depot.config`. The installed manifest is used to compare manifests. We will attempt to download the old manifest if possible.

**Reply by GGHostDotGames** [reply]
> DepotDownloader stores the currently installed manifest ID in `.DepotDownloader/depot.config`. The installed manifest is used to compare manifests. We will attempt to download the old manifest if possible.

Is there a way to retrieve this currently installed manifest ID? The depot.config file seems obfuscated.

**Reply by SlejmUr** [reply]
depotId_manifestId.bin, and the config is compressed.

**Reply by GGHostDotGames** [reply]
> depotId_manifestId.bin

depotId_manifestId.bin shows the latest available config, not the installed config. So if i run -manifest-only, it will change the depotId_manifestId.bin file, even if that is not the installed manifest.

Is there a way to uncompress the config file?
